xmtp/src/client.rs

```diff
@@ -360,6 +365,24 @@ where
 
         Ok(response.envelopes)
     }
+
+    /// Search network for a specific InstallationContact
+    /// This function should be removed as soon as possible given it is a potential DOS vector.
+    /// Contacts for a message should always be known to the client
+    pub async fn download_contact_for_installation(
```

Comment:
If the first message received by the client is a PreKey payload which is not an invite, then the client doesn't have the necessary contact bundle to create a valid session. This results in a linear search through the contact bundles

Reply:
Would it help to have the sender contact attached to the pre-key message?

Reply:
Yes. I am mulling over some larger changes in the protos. Namely the addition of a "Payload" type which unifies Invites and Messages -- There seems to be a lot of overlap between these.

The payload type would contain the same message header from PadlockMessages, and integrate the contact bundle as needed. That way the processing for all packets could be the same, albeit with different contents.

Reply:
Excellent, I was wondering if we should do that too :+1: